src/xmppjs/GatewayMUCMembership.ts

```diff
@@ -49,11 +73,14 @@ export class GatewayMUCMembership {
     }
 
     public getMemberByAnonJid<G extends IGatewayMember>(chatName: string, anonJid: string): G|undefined {
-        return this.getMembers(chatName).find((user) => user.anonymousJid.toString() === anonJid) as G;
+        return Array.from(this.getMembers(chatName)).find((user) => user.anonymousJid.toString() === anonJid) as G;
```

Comment:
I can't believe JS doesn't support find/filter on iterators – straight up encouraging you to create this sort of giant Arrays just to throw them away a moment later.

I didn't add more indexes to ChatMembers for these though, since I was unable to make these functions show up in the profiler output at all (it drops everything that didn't exceed 1% of its parent's runtime).

Reply:
The 1% is fine but yeah the inability to iterate on sets is, cough, awful. Almost tempted to use a utility library...

Reply:
Yeah, I was looking for something in our existing dependency chain, but nothing caught me eye. Someday... :)